cloudprofiler/profiler.go

```diff
@@ -29,26 +30,72 @@ func Start(config Config) error {
 	if !config.Enabled {
 		return nil
 	}
+
+	var svcConfig profiler.Config
+
+	// cloudruntime.Service() always returns true so we query
+	// the environment variable directly for Cloud Run Services.
+	if _, ok := os.LookupEnv("K_SERVICE"); ok {
+		var err error
+		svcConfig, err = cloudRunServiceConfig()
+		if err != nil {
+			return fmt.Errorf("start profiler: %w", err)
+		}
+	} else if _, ok := cloudruntime.Job(); ok {
+		var err error
+		svcConfig, err = cloudRunJobConfig()
+		if err != nil {
+			return fmt.Errorf("start profiler: %w", err)
+		}
+	}
```

Comment:
IMO it would be nice to keep the environment variable handling contained in `cloudruntime`. I wonder if we could change `cloudruntime.Service()` in a compatible and feasible way so that it only returns true for Cloud Run services and not jobs?

If not, a small suggestion to avoid looking at the env var by turning this check around (and using `else` to keep previous behavior of returning error when `K_SERVICE` is not set). But this is not necessary, use as you will! 😄 
```suggestion
	// cloudruntime.Service() always returns true so check
	// cloudruntime.Jobs() first to deduce if it's a Job or not.
	if _, ok := cloudruntime.Job(); ok {
		var err error
		svcConfig, err = cloudRunJobConfig()
		if err != nil {
			return fmt.Errorf("start profiler: %w", err)
		}
	} else {
		var err error
		svcConfig, err = cloudRunServiceConfig()
		if err != nil {
			return fmt.Errorf("start profiler: %w", err)
		}

	}
```